docs(sparkplug): added documentation for SparkplugSubscriber (#5120)
eclipse/kura: docs/cloud-platform/kura-sparkplug.md

##### Sparkplug Device Payload

> The payload of the sent `DDATA` message will be encoded using the [Sparkplug B Protobuf definition](https://github.com/eclipse/tahu/blob/3.x/sparkplug_b/sparkplug_b.proto) converting the [`KuraPayload`](https://github.com/eclipse/kura/blob/develop/kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/message/KuraPayload.java) into [Sparkplug payload](https://github.com/eclipse-sparkplug/sparkplug/blob/3.x/specification/src/main/asciidoc/chapters/Sparkplug_6_Payloads.adoc#payload) as follows:

Reviewer: ```suggestion
The payload of the `DDATA` message to be sent will be encoded using the [Sparkplug B Protobuf definition](https://github.com/eclipse/tahu/blob/3.x/sparkplug_b/sparkplug_b.proto) converting the [`KuraPayload`](https://github.com/eclipse/kura/blob/develop/kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/message/KuraPayload.java) into [Sparkplug payload](https://github.com/eclipse-sparkplug/sparkplug/blob/3.x/specification/src/main/asciidoc/chapters/Sparkplug_6_Payloads.adoc#payload) as follows:
```

Maybe we can also remove the "to be sent". Seems redundant

- [`KuraPayload.getTimestamp()`](https://github.com/eclipse/kura/blob/d53ec833b7438a70a0e3a79406f4c8aed52e94f0/kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/message/KuraPayload.java#L76), if not null, it will be used as the timestamp metric of the Sparkplug payload

#### Sparkplug Subscriber

> This cloud connections allows creating a Cloud Subscriber for subscribing to a generic set of topics. The configuration is shown in the picture below.

Reviewer: ```suggestion
This cloud connections allows creating a Cloud Subscriber that will subscribe to a generic set of topics. The configuration is shown in the picture below.
```

![](./images/sparkplugSubscriber.png)

It is assumed that the payloads received by this Cloud Connection are encoded using the [Sparkplug B Protobuf definition](https://github.com/eclipse/tahu/blob/3.x/sparkplug_b/sparkplug_b.proto). Users of this Cloud Subscriber should expect to receive `KuraMessage`s containing a `KuraPayload` such that:

- If non null, the [Sparkplug body](https://github.com/eclipse-sparkplug/sparkplug/blob/3.x/specification/src/main/asciidoc/chapters/Sparkplug_6_Payloads.adoc#payload) is used to set [`KuraPayload.setBody()`](https://github.com/eclipse/kura/blob/1064870aad5d6f344864644e8c7dadc9ae5d5697/kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/message/KuraPayload.java#L124)
- If non null, the [Sparkplug seq Metric](https://github.com/eclipse-sparkplug/sparkplug/blob/3.x/specification/src/main/asciidoc/chapters/Sparkplug_6_Payloads.adoc#payload) is converted into a [Kura metric](https://github.com/eclipse/kura/blob/1064870aad5d6f344864644e8c7dadc9ae5d5697/kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/message/KuraPayload.java#L96) with name `seq`

> - If non null, the [Sparkplug timestamp](https://github.com/eclipse-sparkplug/sparkplug/blob/3.x/specification/src/main/asciidoc/chapters/Sparkplug_6_Payloads.adoc#payload) is used as Kura's payload [timestamp](https://github.com/eclipse/kura/blob/1064870aad5d6f344864644e8c7dadc9ae5d5697/kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/message/KuraPayload.java#L80)

Reviewer: ```suggestion
- If non null, the [Sparkplug timestamp](https://github.com/eclipse-sparkplug/sparkplug/blob/3.x/specification/src/main/asciidoc/chapters/Sparkplug_6_Payloads.adoc#payload) is used as Eclipse Kura's payload [timestamp](https://github.com/eclipse/kura/blob/1064870aad5d6f344864644e8c7dadc9ae5d5697/kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/message/KuraPayload.java#L80)
```

- All [Sparkplug Metrics](https://github.com/eclipse-sparkplug/sparkplug/blob/3.x/specification/src/main/asciidoc/chapters/Sparkplug_6_Payloads.adoc#payload) are converted into [Kura metrics](https://github.com/eclipse/kura/blob/1064870aad5d6f344864644e8c7dadc9ae5d5697/kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/message/KuraPayload.java#L96) with the same name and the following conversion rules:
    
    | Sparkplug ValueCase | Java Type |
    | - | - |
    | `Boolean` | `boolean` |
    | `Bytes` | `byte[]` |
    | `Dataset` | `byte[]` |
    | `Double` | `double` |
    | `Extension` | `byte[]` |
    | `Float` | `float` |
    | `Integer` | `int` |
    | `Long` | `long` |
    | `String` | `String` |
    | `Template` | `byte[]` |
    | default | `null` |

>     Timestamp of the metric is not supported in Kura, therefore this information is lost at conversion.

Reviewer: ```suggestion
    The metric timestamp is not supported in Eclipse Kura, therefore this information is lost at conversion.
```
